### ISSUE TITLE
Less pandas, more duckdb ;)

### ISSUE THREAD

**Issue by simonaubertbd** [amphi-ai/amphi-etl#214]
Hello,

Regarding performance, shouldn't we use a lot more pandas like explained here :
https://duckdb.org/docs/guides/python/sql_on_pandas.html
https://duckdb.org/2021/05/14/sql-on-pandas.html

I think at least for aggregations, etc.. 

Best regards,

Simon

**Comment by DSandovalFlavio** [comment]
I think that post exemplifies how DuckDB has better performance than pandas

**Comment by tgourdel** [comment]
Yes, it was a dilemma for me to choose. I used pandas for all 'primitives' for simplicity. I was afraid that mixing pandas and DuckDB would have resulted in messy code. However, I now recognize that the performance gains might outweigh that disadvantage.

**Comment by simonaubertbd** [comment]
@tgourdel  What I have in mind would be a default engine (let's say duckdb)  and a dropdown menu that let the user choose on the upper side (since we can imagine the following options can vary according to the engine) wherever it can be performed by several engines. like
 🐼 Pandas  🐼
 🦆 DuckDB 🦆

**Comment by simonaubertbd** [comment]
@tgourdel With my last reading these weeks, I actually believe we could have the following :
-in-memory engine : choice between pandas, duckdb and polars
-in-database : dedicated tools and pure sql

But maybe github isn't the right place to discuss it?
